### PR TITLE
Wrong behavior for nan converting from float to int

### LINKED ISSUE #1761: Wrong behavior for nan converting from float to int
When changing the dtype of an SArray from float to int, `nan` get converted to very large number:
```python
In [1]: import turicreate as tc

In [2]: sa = tc.SArray([1.0,2.0,float('nan'),3.0])

In [3]: sa
Out[3]: 
dtype: float
Rows: 4
[1.0, 2.0, nan, 3.0]

In [4]: sa.astype(int)
Out[4]: 
dtype: int
Rows: 4
[1, 2, -9223372036854775808, 3]
```
I'm not certain what the best behavior would be here; it seems `nan` is a floating point only representation. It would probably be best to turn the `nan` into `None`. Raising an exception is also probably reasonable.

### PR DESCRIPTION
**Pull request by dhivyaaxim**
This PR resolves #1761 
Wrong behavior for nan converting from float to int

### CODE REVIEW

File: src/python/turicreate/test/test_sarray.py [hunk 1]
```diff
@@ -252,6 +252,11 @@ def test_list_with_none_creation(self):
         self.assertEqual(len(g), len(tlist))
         for i in range(len(tlist)):
           self.assertEqual(g[i], tlist[i])
+    
+    def test_nan_to_none_type_conversion(self):
+        sa=SArray([1.0,2.0,float('nan'),3.0])
+        sa=sa.astype(int)
+        self.assertTrue(None in sa)
```
Comment (TobyRoseman) [inline src/python/turicreate/test/test_sarray.py]: This is good but it's better to to check that there is only one `None` and that it's where it's suppose to be. Also good to check the other elements are what we expect too. 

I would change this line to:
```python
expected = SArray([1, 2, None, 3])
for a, b in zip(sa, expected):
    self.assertEqual(a, b)
```

**Comment by dhivyaaxim** [comment]
@TobyRoseman We had added changes from previous fork turicreate master and run unit test for python and C, it's got failed. Pulled master changes and integrate my changes, then run unit test, it's passed. Any possible to sync latest master changes and integrate my changes in this commit? Please advise. Thanks in advance!

**Comment by TobyRoseman** [comment]
@dhivyaaxim - These error are unrelated to your change. I'll just merge this now.

**Comment by dhivyaaxim** [comment]
@TobyRoseman Thanks for your prompt response! Please merge and close the below two issues too have similar build issue
**#954** and **#2390**

**Pull request merged**